Yosemite/YosemiteTests/Stores/PaymentGatewayStoreTests.swift

```diff
@@ -83,4 +92,73 @@ final class PaymentGatewayStoreTests: XCTestCase {
+        // Given
+        let samplePaymentGatewayID = "cod"
+        let samplePaymentGateway = PaymentGateway.fake().copy(siteID: sampleSiteID,
+                                                              gatewayID: samplePaymentGatewayID,
+                                                              title: "Failing gateway",
+                                                              enabled: false)
+        storePaymentGateway(samplePaymentGateway, for: sampleSiteID)
+        assertEqual(1, storedPaymentGatewaysCount)
+
+        let expectedError = NetworkError.unacceptableStatusCode(statusCode: 500)
+        network.simulateError(requestUrlSuffix: "payment_gateways/\(samplePaymentGatewayID)", error: expectedError)
+
+        // When
+        let updatedPaymentGateway = samplePaymentGateway.copy(title: "Cash on delivery")
+        let result: Result<Networking.PaymentGateway, Error> = waitFor { promise in
+            let action = PaymentGatewayAction.updatePaymentGateway(updatedPaymentGateway) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then
+        XCTAssertEqual(result.failure as? NetworkError, expectedError)
+        assertEqual(1, storedPaymentGatewaysCount)
+        assertEqual("Failing gateway", viewStorage.loadPaymentGateway(siteID: sampleSiteID, gatewayID: samplePaymentGatewayID)?.title)
+    }
+
+    func test_updatePaymentGateway_updates_stored_paymentGateway_upon_success() throws {
+        // Given
+        let samplePaymentGatewayID = "cod"
+        let samplePaymentGateway = PaymentGateway.fake().copy(siteID: sampleSiteID,
+                                                              gatewayID: samplePaymentGatewayID,
+                                                              title: "Old title",
+                                                              enabled: false)
+
+        storePaymentGateway(samplePaymentGateway, for: sampleSiteID)
+        assertEqual(1, storedPaymentGatewaysCount)
+
+        network.simulateResponse(requestUrlSuffix: "payment_gateways/\(samplePaymentGatewayID)", filename: "payment-gateway-cod")
+
+        // When
+        let updatedPaymentGateway = samplePaymentGateway.copy(enabled: true)
+        let result: Result<Networking.PaymentGateway, Error> = waitFor { promise in
+            let action: PaymentGatewayAction
+            action = .updatePaymentGateway(updatedPaymentGateway) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+
+        let storedPaymentGateway = try XCTUnwrap(viewStorage.loadPaymentGateway(siteID: sampleSiteID, gatewayID: samplePaymentGatewayID))
+        assertEqual("Cash on delivery", storedPaymentGateway.title)
+        XCTAssertTrue(storedPaymentGateway.enabled)
+    }
+}
+
+private extension PaymentGatewayStoreTests {
+    @discardableResult
```

Comment:
From what I understand we use `@discardableResult` here as we don't have to use its return for anything else than counting in the test assertion, and we can hide the different warnings that this causes as unused, is that right? 

```
assertEqual(1, storedPaymentGatewaysCount)
```

Reply:
Correct!